### DISCUSSION TITLE
Is there a way to delay configuration of the Router?

### DISCUSSION THREAD

**Discussion by tolmanam** [Lancetnik/Propan#86] [Questions] [answered]
I don't know the connection URL for my message bus until after I've had to chance to parse my configuration file and user input.  Is there a way to delay the configuration of the router?

I'm wondering if we might be able to do something clever with 'top-level dependencies'.

Alternatively, `fast-api` exposes the `add_api_route` that provides a way to add routes without using the decorator so I can specifically set them up after my configuration processing is complete. (this is a bit of a hack, so being able to delay the initialization of the Route may be preferable)

**Comment by Lancetnik** [comment, marked as answer]
@tolmanam sure, you can specify all connection arguments at `broker.connect()` method

Example from documentation:

```python
from propan import RedisBroker
broker = RedisBroker()
...
await broker.connect("redis://localhost:6379/")
```
Also, please, generate any **Propan** template by the command:

```bash
propan create async ... project
```

At `project/app/serve.py` you will find a configuration from `.env` file example in `app.on_startup` hook

**Reply by tolmanam** [reply]
Thank you!
